src/dataset-builder/ConceptSelector.ts

```diff
@@ -83,7 +83,8 @@ export const ConceptSelector = (props: ConceptSelectorProps) => {
           { name: 'Concept ID', width: 195, render: _.get('id') },
           { name: 'Roll-up count', width: 205, render: _.get('count') },
         ],
-        root: rootConcept,
+        root: { id: parents[0].parentId, name: 'root', count: 0, hasChildren: true },
```

Comment:
Why is parents a list? Is it because the concept exists in multiple locations so it has multiple parents? If so, why do we just take the first?

Reply:
Yes, a concept can have multiple parents. But the reason why `parents` is a list is because the tree structure is stored as a flattened set of parent-child relationships, grouped by parent. So each parent entry is a single parent and all of its children.

The reason why it takes the first for the root is because the first parent in the response is always the root of the tree. I went back and forth on this part of the API. Another approach would be to return the parents in any order and the UI code to could find the root by finding the ID of a concept that's in `parents` but is not in any parent's children. Another way would be to have the API return both the list of parents and the root parent ID (or the index). This seemed like the easiest way at the time but I could be convinced to change it.

Reply:
I'll add a comment to `TreeGrid` where `Parent` is defined.

Also I've updated the code so that the API no longer moves the root to the first element, and instead the UI code must find the root after it gets the response back.